src/utils/transaction/insertTransaction.ts

```diff
@@ -15,141 +16,225 @@ import { reportUsage } from "../usage";
+ */
+const transformV4SmartBackendWallet = async (
+  transaction: QueuedTransaction,
+  walletDetails: SmartBackendWalletDetails, // Note: narrowed type
+): Promise<QueuedTransaction> => {
+  await validateSmartBackendWalletChainSupport(transaction.chainId);
 
-  let walletDetails: ParsedWalletDetails | undefined;
+  return {
+    ...transaction,
+    entrypointAddress: walletDetails.entrypointAddress ?? undefined,
+    accountFactoryAddress: walletDetails.accountFactoryAddress ?? undefined,
+    walletType: walletDetails.type,
+  };
+};
 
+/**
+ * Try to resolve wallet details, determining if we're in V4 or V5 case
+ * For V5: wallet details should be found from 'from' address (Cases 1 & 3)
+ * For V4: wallet details are found from accountAddress (Cases 2 & 4)
+ */
+const resolveWalletDetails = async (
+  transaction: QueuedTransaction,
+): Promise<ResolvedWalletDetails> => {
+  // Try V5 path first (Cases 1 & 3)
   try {
-    walletDetails = await getWalletDetails({
-      address: queuedTransaction.from,
+    const walletDetails = await getWalletDetails({
+      walletAddress: transaction.from,
     });
+    return { sdkVersion: "v5", walletDetails };
+  } catch {} // Silently handle V5 failure
 
-    // when using the v5 SDK with smart backend wallets, the following values are not set correctly:
-    // isUserOp is set to false
-    // account address is blank or the user provided value (this should be the SBW account address)
-    // from is set to the SBW account address (this should be the SBW signer address)
-    // these values need to be corrected so the worker can process the transaction
-    if (isSmartBackendWallet(walletDetails)) {
-      if (queuedTransaction.accountAddress) {
-        throw createCustomError(
-          "Smart backend wallets do not support interacting with other smart accounts",
-          StatusCodes.BAD_REQUEST,
-          "INVALID_SMART_BACKEND_WALLET_INTERACTION",
-        );
-      }
-
-      if (
-        !(await doesChainSupportService(
-          queuedTransaction.chainId,
-          "account-abstraction",
-        ))
-      ) {
-        throw createCustomError(
-          "Chain does not support smart backend wallets",
-          StatusCodes.BAD_REQUEST,
-          "SBW_CHAIN_NOT_SUPPORTED",
-        );
-      }
-
-      queuedTransaction = {
-        ...queuedTransaction,
-        isUserOp: true,
-        signerAddress: walletDetails.accountSignerAddress,
-        from: walletDetails.accountSignerAddress,
-        accountAddress: queuedTransaction.from,
-        target: queuedTransaction.to,
-        accountFactoryAddress: walletDetails.accountFactoryAddress ?? undefined,
-        entrypointAddress: walletDetails.entrypointAddress ?? undefined,
-      };
-    }
+  // If primary address fails and no accountAddress, we can't proceed
+  if (!transaction.accountAddress) {
+    throw createCustomError(
+      "Account not found",
+      StatusCodes.BAD_REQUEST,
+      "ACCOUNT_NOT_FOUND",
+    );
```

Comment:
Is there a clearer error here? Is this a use case the user can get into, or does it indicate a bug within Engine (5xx response)?